### ISSUE TITLE
[BUG]: dynamo-run --router-mode kv stateful, doesn't update

### ISSUE THREAD

**Issue by alec-flowers** [ai-dynamo/dynamo#1232]
### Describe the Bug

When you use dynamo-run in=http out=dyn --router-mode
You are able to start and stop models and they are dynamically added to the HTTP server and removed. But if you change a hyperparameter like kv_block_size, because the kv router is initialized only once with the block_size of the first engine it will stop functioning. 

I also don't know if tokenizers fall into this category as well (ie its loaded at the start and then even if we change the model on the python size

### Steps to Reproduce

**Terminal 1:** `dynamo-run in=http out=dyn --router-mode kv`
**Terminal 2:** 
`python vllm_inc.py --kv-block-size 32`
Let it load, send two request, ctrl + c (note that it properly shows routing logits)
`python vllm_inc.py --kv-block-size 64`
Let it load, send two requests, note that it no longer has routing logits (they are all 0 because the indexer can't index blocks with different size kv)

### Expected Behavior

I expected things to just work. So in this case the KvRouter would get reloaded with the correct block size. 

### Actual Behavior

The KvIndexer was initialized once and becomes useless for new engines. 

### Environment

System Information:
OS: Ubuntu 24.04
Glibc Version: 2.39
GCC Version: 13.3.0
Cmake Version: 3.28.3
Rust Version: 1.87.0
Docker Version: N/A

CPU Information:
Architecture:                         x86_64
CPU op-mode(s):                       32-bit, 64-bit
Address sizes:                        46 bits physical, 48 bits virtual
Byte Order:                           Little Endian
CPU(s):                               24
On-line CPU(s) list:                  0-23
Vendor ID:                            GenuineIntel
Model name:                           Intel(R) Core(TM) i9-10920X CPU @ 3.50GHz
CPU family:                           6
Model:                                85
Thread(s) per core:                   2
Core(s) per socket:                   12
Socket(s):                            1
Stepping:                             7
CPU(s) scaling MHz:                   31%
CPU max MHz:                          4800.0000
CPU min MHz:                          1200.0000
BogoMIPS:                             6999.82
Flags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 cdp_l3 ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm mpx rdt_a avx512f avx512dq rdseed adx smap clflushopt clwb intel_pt avx512cd avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local dtherm ida arat pln pts hwp hwp_act_window hwp_epp hwp_pkg_req vnmi avx512_vnni md_clear flush_l1d arch_capabilities
Virtualization:                       VT-x
L1d cache:                            384 KiB (12 instances)
L1i cache:                            384 KiB (12 instances)
L2 cache:                             12 MiB (12 instances)
L3 cache:                             19.3 MiB (1 instance)
NUMA node(s):                         1
NUMA node0 CPU(s):                    0-23
Vulnerability Gather data sampling:   Mitigation; Microcode
Vulnerability Itlb multihit:          KVM: Mitigation: VMX disabled
Vulnerability L1tf:                   Not affected
Vulnerability Mds:                    Not affected
Vulnerability Meltdown:               Not affected
Vulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable
Vulnerability Reg file data sampling: Not affected
Vulnerability Retbleed:               Mitigation; Enhanced IBRS
Vulnerability Spec rstack overflow:   Not affected
Vulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl
Vulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop
Vulnerability Srbds:                  Not affected
Vulnerability Tsx async abort:        Mitigation; TSX disabled

Python Version: 3.12.3
Python Platform: Linux-6.8.0-58-generic-x86_64-with-glibc2.39

Python Packages:
ai-dynamo: 0.2.1
ai-dynamo-runtime: 0.2.1
ai-dynamo-vllm: 0.8.4.post1
genai-perf: Not installed
nixl: 0.2.1
numpy: 1.26.4
nvidia-cublas-cu12: 12.4.5.8
nvidia-cuda-cupti-cu12: 12.4.127
nvidia-cuda-nvrtc-cu12: 12.4.127
nvidia-cuda-runtime-cu12: 12.4.127
nvidia-cudnn-cu12: 9.1.0.70
nvidia-cufft-cu12: 11.2.1.3
nvidia-curand-cu12: 10.3.5.147
nvidia-cusolver-cu12: 11.6.1.9
nvidia-cusparse-cu12: 12.3.1.170
nvidia-ml-py: 12.575.51
nvidia-nccl-cu12: 2.21.5
nvidia-nvjitlink-cu12: 12.4.127
nvidia-nvtx-cu12: 12.4.127
pyzmq: 26.4.0
tensorrt_llm: Not installed
torch: 2.6.0
torchaudio: 2.6.0
transformers: 4.52.3
tritonclient: 2.53.0

**Comment by alec-flowers** [comment]
I'm not actually sure what behavior we want in this situation. I ran into this because I was treating the dynamo-run in=http as static and then doing some debugging and changes and setting up and tearing down the python workers. 

It may be fine the way it is, but we should have some warning or something that lets the user know incompatible blocks are hitting the indexer. I was just seeing silent failure.

**Comment by grahamking** [comment]
Nice catch! We should probably:
- Warn if two instances for the same endpoint have a different block size
- Tear down the router when the last instance goes away.

**Comment by grahamking** [comment]
Auto-closer is annoying.
